### ISSUE TITLE
Predicted Values all zero

### ISSUE THREAD

**Issue by shaleenanuj** [MehrdadNoori/Attention-Guided-Version-of-2D-UNet-for-Automatic-Brain-Tumor-Segmentation#14]
I trained the model and after 22 epoch I tried to predict but the file which come on prediction is all zeros. can you please tell me what is the issue.

**Comment by zhangshuang317** [comment]
I encountered the same situation, did you solve this problem?

**Comment by gyakuz** [comment]
> 
Have you solved it yet,The predict file is not segmented

**Comment by JiahaoHuang99** [comment]
I met the same problem. Have you solved it?

**Comment by JiahaoHuang99** [comment]
> > 
> 
> Have you solved it yet,The predict file is not segmented

I found the result changed when I removed the normalize_volume() in predict.py, but the result is still terrible.

You can try it. And if you trained a good model, could you please send me your weight? Thanks.

**Comment by razeineldin** [comment]
Hi @shaleenanuj, @zhangshuang317, @gyakuz, @JiahaoHuang99,

I faced the same issue and solved it as follows:

The normalization code in [normalize_volume()](https://github.com/Mehrdad-Noori/Brain-Tumor-Segmentation/blob/ebfd12d666dfd6a1743bc4935c6f68a8e26370e3/predict.py#L88) in predict.py has to be the same as the training normalization code.

`normalized_slices[..., slice_ix] = normalize_slice(input_volume[..., slice_ix])`

Instead of 

`normalized_slices[slice_ix] = input_volume[slice_ix]`
`for mode_ix in range(input_volume.shape[1]):`
`     normalized_slices[slice_ix][mode_ix] = normalize_slice(input_volume[slice_ix][mode_ix])`

**Comment by sneh-debug** [comment]
@shaleenanuj Please tell me the dependency of tensorflow and keras you have installed to run this code?

**Comment by Coolwen1997** [comment]
> > > 
> > 
> > 
> > Have you solved it yet,The predict file is not segmented
> 
> I found the result changed when I removed the normalize_volume() in predict.py, but the result is still terrible.
> 
> You can try it. And if you trained a good model, could you please send me your weight? Thanks.

Sorry to bother you. The segmentation results from the trained model are terrible.  And the val_loss is close to 0.9 even in the 30th epoch when I am training. Have you occurred the same problem and how did you solved it ? Thanks
